### PR TITLE
Use a method to get CMSApp.urls, to allow passing a page object to it

### PR DESCRIPTION
**Pull request by yakky**
This a proposal to change the CMSApp a bit, to allow defining urlconfs base on the page the apphook is attached to
As of now code does not work :), but I'm interested on the feedback over the general idea

### CODE REVIEW

**Comment by yakky** [comment]
Ping @divio/django-cms-core for comments

**Comment by stefanfoulis** [comment]
@yakky great idea. What are some example use-cases?

**Comment by yakky** [comment]
@stefanfoulis when an application needs to use different urlconf when apphooks-config / it's attached multiple times : blog uses a set of urlconfs, news a different one. By  checking the instance_name the "concrete" get_urls() method of the apphook can provide a different set of urls.
The method is obviously meant to be redefined in child classes to use the page argument.

**Comment by yakky** [comment]
@divio/django-cms-core any more comment about this?

File: cms/app_base.py [hunk 1]
```diff
@@ -32,3 +43,17 @@ def get_config(self, namespace):
...
+        warnings.warn('Accessing CMSApp.urls directly is deprecated, '
+                      'and it will be removed in version 3.5; CMSApp.get_urls method',
+                      DeprecationWarning)
+        return self._urls
+
+    @legacy_urls.setter
+    def urls(self, value):
+        self._urls = value
+
+    def get_urls(self, page=None):
```
Comment (mkoistinen) [inline cms/app_base.py]: Since this is all about flexibility, perhaps it makes sense to include `**kwargs` here?

**Comment by mkoistinen** [comment]
I'm a fan of the concept and the implementation. I made a small suggestion too. =)

**Comment by yakky** [comment]
@stefanfoulis @mkoistinen one step beyond. At this point only tests should be missing

**Comment by yakky** [comment]
@divio/django-cms-core complete on my side, please review

**Comment by mkoistinen** [comment]
@yakky any chance this can get passing tests and another approval asap?

**Comment by czpython** [comment]
Ok to merge

**Pull request merged**